includes/Modules/AdSense.php

```diff
@@ -537,9 +537,17 @@ protected function create_adsense_earning_data_request( array $args = array() )
 			$opt_params['maxResults'] = (int) $args['limit'];
 		}
 
-		$host = wp_parse_url( $this->context->get_reference_site_url(), PHP_URL_HOST );
-		if ( ! empty( $host ) ) {
-			$opt_params['filter'] = 'DOMAIN_NAME==' . $host;
+		$hostnames = array_unique(
+			array_map(
+				function ( $site_url ) {
+					return 'DOMAIN_NAME==' . wp_parse_url( $site_url, PHP_URL_HOST );
+				},
+				$this->permute_site_url( $this->context->get_reference_site_url() )
+			)
+		);
+
+		if ( ! in_array( '', $hostnames, true ) ) {
```

Comment:
This condition is unnecessary because it would never contain an empty string.
https://github.com/google/site-kit-wp/blob/1680f18750ea7fc778d39a5a5feab64d0d1b24d0/includes/Core/Modules/Module.php#L645-L652